hooks/command

```diff
@@ -186,15 +186,22 @@ else
   while true
   do
     set +e
-    pod_json=$(kubectl get pod "$pod_name" -o json)
-    init_container_status="$(echo "$pod_json" | jq ".status.initContainerStatuses[0].state.terminated.exitCode")"
-    if [[ -n "$init_container_status" && "$init_container_status" != "0" ]]; then
-      echo "Warning: init container failed with exit code $init_container_status, this usually indicates plugin misconfiguration or infrastructure failure"
-      status="$init_container_status"
-    else
-      status="$(echo "$pod_json" | jq ".status.containerStatuses[0].state.terminated.exitCode")"
-    fi
+    pod_json=$(kubectl get pod "$pod_name" -o json 2>&1)
+    get_pod_status=$?
     set -e
+    if [[ "$get_pod_status" -eq 0 ]]; then
+      init_container_status="$(echo "$pod_json" | jq ".status.initContainerStatuses[0].state.terminated.exitCode")"
+      if [[ -n "$init_container_status" && "$init_container_status" != "0" ]]; then
+        echo "Warning: init container failed with exit code $init_container_status, this usually indicates plugin misconfiguration or infrastructure failure"
+        status="$init_container_status"
+      else
+        status="$(echo "$pod_json" | jq ".status.containerStatuses[0].state.terminated.exitCode")"
+      fi
+    elif [[ "$pod_json" == *NotFound* ]]; then
+      echo "Warning: pod not found"
+      status="42"
+    fi
```

Comment:
we also need an else here to print out the content of pod_json which should contain stderr at this point - so that we don't swallow the error message if the command failed for any other reason than 'NotFound'

Reply:
Since we're in this loop could that end up printing noisy, potentially expected, failures when we'd actually just want the loop to kick in and try again?

Reply:
Can we add an else block that prints out what the error message was? thinking it would be nice to see in the logs what the issue is rather than the script failing silently

Reply:
ah yea, I guess we would have seen this before as well so it shouldn't be any worse than today, added